File: gcj2013R1AA.py
#!/usr/bin/env python
import sys

def solve():
    r,t = map(int,sys.stdin.readline().split())
    lo,hi = 0, r+t+1
    while lo+1<hi:
        m = (lo+hi)/2
        if 2*m*r+2*m*m-m>t: hi=m
        else: lo=m
    return lo

for x in range(int(sys.stdin.readline())):
    print("Case #"+str(x+1)+": "+str(solve()))

File: gcj2013R1AA.hs
#!/usr/bin/env runghc
main = getContents >>= mapM_ putStrLn . zipWith caseno [1..] . solve . lines
caseno i r = "Case #"++show i++": "++r
solve (c:cs) = map (toString . answer) . take (read c) . analyze $ cs ::[String]
analyze = map (map read . words)
toString = show
answer [r,t] = f 0 (r+t+1) where
  f lo hi
    | lo+1>=hi = lo
    | 2*m*r+2*m*m-m>t = f lo m
    | otherwise = f m hi
    where m = (lo+hi)`div`2
